analysis/deciles_charts.py

```diff
@@ -81,17 +102,24 @@ def parse_args():
         type=get_path,
         help="Path to the output directory",
     )
+    parser.add_argument(
+        "--config",
+        default=DEFAULT_CONFIG.copy(),
```

Comment:
Great question! If we didn't copy, then the `dict` that's `args.config` (the value of `default`) would be the same `dict` that's `DEFAULT_CONFIG`. If we were to change `args.config`, then we would also change `DEFAULT_CONFIG`, which is misleading: `DEFAULT_CONFIG` would no longer be the default configuration. Consider the following, which I'd encourage you to copy-and-paste into an IPython shell:

```python
>>> import argparse
>>> DEFAULT_CONFIG = {"my_key": "my_value"}
>>> parser = argparse.ArgumentParser()
>>> parser.add_argument("--config", default=DEFAULT_CONFIG)
>>> args = parser.parse_args()
>>> args.config
{'my_key': 'my_value'}
>>> args.config is DEFAULT_CONFIG
True
>>> args.config["my_key"] = "my_new_value"
>>> args.config
{'my_key': 'my_new_value'}
>>> DEFAULT_CONFIG
{'my_key': 'my_new_value'}
```

Unlike, say, `tuple`s, `dict`s are mutable; the standard library doesn't contain an immutable `dict`, either. We indicate that this variable is a constant by using uppercase (`DEFAULT_CONFIG` rather than `default_config`) but this is a convention.

Reply:
Thanks, that makes sense. And now, so do the uppercase variables i've seen in various places 😃

Reply:
Thanks, that makes sense. And now, so do the uppercase variables i've seen in various places 😃

Reply:
Yup, uppercase is used to indicate that the variable is a constant, and so should be considered immutable. As an aside, notice that classes are [upper-camel case](https://en.wikipedia.org/wiki/Camel_case) e.g. `MyClass`.